lib/Sema/TypeCheckType.cpp

```diff
@@ -879,8 +874,9 @@ Type TypeChecker::applyUnboundGenericArguments(
         return BoundGenericType::get(nominalDecl, parentType, genericArgs);
       }
 
-      assert(!resultType->hasTypeParameter());
-      return resultType;
+      const auto resultTy = decl->getDeclaredInterfaceType();
+      assert(!resultTy->hasTypeParameter());
```

Comment:
This assertion will fail if decl is generic

Reply:
I wonder what this is trying to accomplish ignoring the arguments. This is the code path for Foo.Alias<...>, where Foo is an unbound generic parent. I will see if I can reproduce this assertion failure once I get the Standard Library to build once again.